src/main/scala/com/fulcrumgenomics/sopt/cmdline/CommandLineProgramParser.scala

```diff
@@ -227,19 +227,34 @@ class CommandLineProgramParser[T](val targetClass: Class[T], val includeSpecialA
       .filter { withSpecial || !_.isSpecial }
       .partition { !_.optional }
 
-    if (required.nonEmpty) {
-      builder.append(KRED(s"\n${KBLDRED(targetName)} ${KRED(RequiredArguments)}\n"))
-      builder.append(lineBreak(color=true))
-      new ClpArgumentLookup(required:_*).ordered.foreach { arg =>
-        ClpArgumentDefinitionPrinting.printArgumentDefinitionUsage(builder, arg, argumentLookup)
+    // Split the arguments by group
+    val groups = argumentLookup.view
+      .filterNot { _.hidden }
+      .filter { withSpecial || !_.isSpecial }
+      .groupBy(_.groupName.getOrElse(""))
+
+    // Print out the commands by group
+    groups.keys.toSeq.sorted.map(_.capitalize).foreach { groupName =>
```

Comment:
I think sorting here is undesirable.  We present options in the order they are declared in the CLP, giving the user control over things.  I think the thing to do here would be to instead iterate over all of the options and add the `group`s to a `LinkedHashSet`.  That way you'd get the unique set of groups, ordered by the first time they showed up in the options.  It's not perfect since you could still have:

```scala
@arg(group="foo") val x,
@arg(group="bar") val y,
@arg(group="foo") val z
```

but it would probably be the least surprising result and give the user some control over things.